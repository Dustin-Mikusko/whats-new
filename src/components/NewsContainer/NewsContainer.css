main {
  padding: 20px;
  height: 100%;
}

.search {
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin-bottom: 20px;
  padding: 0px 20px 0px 20px;
}

.news-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  column-count: 2;
  overflow: scroll;
  height: 100vh;
}

.search-input {
  width: 40%;
  color: white;
  border: none;
  background-color: #141527;
  font-size: 1.5em;
}

.search-btn {
  border-radius: 30px;
  width: 200px;
  background-color: rgb(176, 63, 208);
  color: white;
  font-size: 1.3em;
  border: none;
}

.search-btn:hover {
  background-color: rgb(199, 95, 228);
  cursor: pointer;
}

.search-btn:active {
  transform: scale(.97);
}
