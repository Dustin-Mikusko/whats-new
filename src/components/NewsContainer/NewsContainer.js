import React from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle';
import SearchForm from '../SearchForm/SearchForm';

const NewsContainer = ({ news }) => {
  const newsCards = news.map(article => {
    return (
      <NewsArticle 
        key={article.id}   
        id={article.id}
        title={article.headline}
        description={article.description}
        url={article.url}
        img={article.img}
      />
    )
  })


  return (
    <main> 
      <SearchForm />
      <section className="news-container">
        {newsCards}
      </section>
    </main>
  )
}

export default NewsContainer;
