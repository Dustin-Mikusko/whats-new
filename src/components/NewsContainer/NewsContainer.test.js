import React from 'react';
import NewsContainer from './NewsContainer';
import { shallow } from 'enzyme';

describe('NewsContainer', () => {
  let wrapper;
  let mockNews;

  beforeEach(() => {
    mockNews = [
      {
      id: 1,
      headline: 'The Who postpones Denver Concert at the Pepsi Center',
      img: 'https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg',
      description: 'Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.',
      url: 'https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/'
    }
  ];
    wrapper = shallow(<NewsContainer 
      news={mockNews}
    />
    );
  });

  it('should display NewsContainer with news cards and match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
