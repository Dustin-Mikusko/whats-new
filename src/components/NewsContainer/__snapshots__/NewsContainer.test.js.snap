// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should display NewsContainer with news cards and match snapshot 1`] = `
<main>
  <SearchForm />
  <section
    className="news-container"
  >
    <NewsArticle
      description="Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter."
      id={1}
      img="https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg"
      key="1"
      title="The Who postpones Denver Concert at the Pepsi Center"
      url="https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
    />
  </section>
</main>
`;
